src/core/document.js

```diff
@@ -1713,6 +1713,9 @@ class PDFDocument {
 
   #collectFieldObjects(name, fieldRef, promises, annotationGlobals) {
     const field = this.xref.fetchIfRef(fieldRef);
+    if (!field) {
```

Comment:
This could still cause errors in a corrupt PDF document, please ensure that it's actually a dictionary since the code below currently assumes that.
```suggestion
    if (!(field instanceof Dict)) {
```